orderbook/src/orderbook.rs

```diff
@@ -79,7 +81,15 @@ impl Orderbook {
             return Ok(AddOrderResult::InsufficientFee);
         }
         let order = match Order::from_order_creation(order, &self.domain_separator) {
-            Some(order) => order,
```

Comment:
I'd like to keep the match on from_order_creation and payload separate because it's easier to read:
This can stay as before:

```rust
        let order = match Order::from_order_creation(order, &self.domain_separator) {
            Some(order) => order,
            None => return Ok(AddOrderResult::InvalidSignature),
        };
```
and then we add either
```rust
        if matches!(payload.from, Some(from) if from != order.order_meta_data.owner) {
            return Ok(AddOrderResult::WrongOwner(order.order_meta_data.owner));
        }
```
or
```rust
        if let Some(from) = payload.from {
            if from != order.order_meta_data.owner {
                return Ok(AddOrderResult::WrongOwner(order.order_meta_data.owner));
            }
        }
```
Not sure which one you find clearer.

Reply:
Cool! Thanks.